File: CPP_SOLUTIONS/38.cpp
class Solution {
public:
    string i2s(int i) {
        stringstream ss;
        ss << i;
        string res;
        ss >> res;
        return res;
    }
    string countAndSay(int n) {
        string res = "1";
        for(int i = 1; i < n; i++) {
            int l = 0, r = 0;
            string tmp = "";
            while(l < res.size()) {
                while(r < res.size() && res[l] == res[r]) r ++;
                tmp += (to_string(r - l) + res[l]);
                l = r;
            }
            res = tmp;
        }
        return res;
    }
};

File: PYTHON_SOLUTIONS/38.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        s = "1"
        for i in range(n-1):
            j, n= 0,0
            tmp = ""
            while j < len(s):
                c = 0
                while n < len(s) and s[j] == s[n]:
                    n += 1
                    c += 1
                tmp = tmp + str(c) + s[j]
                j = n
            s = tmp
        return s
        
        
